### PR TITLE
Add Travis CI jobs for Arm and AArch64

### PR DESCRIPTION
**Pull request by kpet**
- Use fixed Linaro toolchains
- Move travis script to a shell file

Signed-off-by: Kevin Petit <kevin.petit@arm.com>

### CODE REVIEW

File: travis.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+TOOLCHAIN_FILE=${TOP}/toolchain.cmake
+touch ${TOOLCHAIN_FILE}
+
+# Prepare toolchain if needed
+if [[ ${JOB_ARCHITECTURE} != "" ]]; then
+    TOOLCHAIN_URL_VAR=TOOLCHAIN_URL_${JOB_ARCHITECTURE}
+    TOOLCHAIN_URL=${!TOOLCHAIN_URL_VAR}
+    wget ${TOOLCHAIN_URL}
+    TOOLCHAIN_ARCHIVE=${TOOLCHAIN_URL##*/}
+    tar xvf ${TOOLCHAIN_ARCHIVE}
```
Comment (Jeremy-Kemp) [inline travis.sh]: Could we drop the v? Its quite a spammer.

**Comment by Jeremy-Kemp** [comment]
Seeing the gcc 7.5 toolchain got me thinking about how to catch new language features being added that might not work on older gcc's.

I'll create an issue where we can start thinking about what minimum versions of gcc we want to support.

**Comment by jrprice** [comment]
Is there a reason these jobs are being done via cross-compilation rather than native `arm64` jobs (which Travis CI apparently supports)?

File: travis.sh (hunk 1 above)
Comment (kpet) [reply travis.sh]: Yeah, it runs quickly enough. I'll remove it.

**Comment by kpet** [comment]
We could do native builds for `arm64` but it seems Travis doesn't support `arm` so we'd still need cross-compilation for that.

**Comment by Jeremy-Kemp** [comment]
We should probably just stick with the script for both `arm` and `arm64` then to avoid introducing additional complexity to the build.

**Comment by kpet** [comment]
Thanks for the review, I'll give @jrprice a chance to comment before merging this.

**Pull request merged**